helix-term/src/commands.rs

```diff
@@ -4050,6 +4055,98 @@ fn surround_delete(cx: &mut Context) {
     })
 }
 
+#[derive(Eq, PartialEq)]
+enum ShellBehavior {
+    Replace,
+    Insert,
+    Append,
+    Filter,
+    None,
+}
+
+fn shell_pipe(cx: &mut Context) {
+    shell(cx, "pipe:", true, ShellBehavior::Replace);
+}
+
+fn shell_pipe_ignore(cx: &mut Context) {
+    shell(cx, "pipe:", true, ShellBehavior::None);
+}
+
+fn shell_insert(cx: &mut Context) {
+    shell(cx, "command:", false, ShellBehavior::Insert);
+}
+
+fn shell_append(cx: &mut Context) {
+    shell(cx, "command:", false, ShellBehavior::Append);
+}
+
+fn shell_filter(cx: &mut Context) {
+    shell(cx, "filter:", true, ShellBehavior::Filter);
+}
+
+fn shell(cx: &mut Context, prompt: &str, pipe: bool, behavior: ShellBehavior) {
+    use std::io::Write;
+    use std::process::{Command, Stdio};
+    let shell = cx.editor.config.shell.clone();
```

Comment:
I'm unsure of how to take the shell config without cloning.

Reply:
I think if you fetch it inside the Prompt's `move` closure then it will allow you to use it by ref.

Reply:
That worked :+1: